src/main/java/org/sagebionetworks/web/client/widget/ReactComponent.java

```diff
@@ -1,180 +1,55 @@
 package org.sagebionetworks.web.client.widget;
 
 import com.google.gwt.dom.client.DivElement;
-import com.google.gwt.dom.client.Document;
-import com.google.gwt.dom.client.Element;
 import com.google.gwt.event.dom.client.ClickEvent;
 import com.google.gwt.event.dom.client.ClickHandler;
 import com.google.gwt.event.dom.client.HasClickHandlers;
 import com.google.gwt.event.shared.HandlerRegistration;
 import com.google.gwt.user.client.Timer;
-import com.google.gwt.user.client.ui.ComplexPanel;
-import com.google.gwt.user.client.ui.Widget;
-import java.util.ArrayList;
-import java.util.List;
-import jsinterop.base.JsPropertyMap;
-import org.sagebionetworks.web.client.jsinterop.React;
-import org.sagebionetworks.web.client.jsinterop.ReactComponentProps;
-import org.sagebionetworks.web.client.jsinterop.ReactComponentType;
+import com.google.gwt.user.client.ui.FlowPanel;
 import org.sagebionetworks.web.client.jsinterop.ReactDOM;
 import org.sagebionetworks.web.client.jsinterop.ReactDOMRoot;
 import org.sagebionetworks.web.client.jsinterop.ReactElement;
 
 /**
- * Widget that manages the lifecycle of a React component. To use this widget, create a {@link ReactElement} using the
- * {@link React} API and call {@link #render(ReactElement)} to render the React component.
- * <p>
- * This widget also manages appending child elements if the associated React component can contain children. If all
- * child widgets use this class, then the child {@link ReactElement}s will be cloned and passed as children to the React
- * component. If any child of this component is a non-ReactComponent widget, then the child widgets will be injected
- * into the node found using the component's `ref`.
- * <p>
- * The root element defaults to a `div`, but can be changed (e.g. to a `span`) using the {@link #ReactComponent(String)} constructor.
- * <p>
- * This widget automatically unmounts the ReactComponent (if any) when this container is detached/unloaded.
+ * Automatically unmounts the ReactComponent (if any) inside this div when this container is detached/unloaded.
  */
-public class ReactComponent<T extends ReactComponentProps>
-  extends ComplexPanel
-  implements HasClickHandlers {
+public class ReactComponent extends FlowPanel implements HasClickHandlers {
```

Comment:
ReactComponent logic was reverted to now match what it was before (in ReactComponentDiv). The only new logic that I kept was the ability to override the tag e.g. to span.

Like ReactComponentDiv before it, this component can no longer hold children, only the V2 version can.